index.js

```diff
@@ -1,46 +1,51 @@
 'use strict';
 
-var Plugin = require('broccoli-plugin');
-var MergeTrees = require('merge-trees');
-
-module.exports = BroccoliMergeTrees;
-BroccoliMergeTrees.prototype = Object.create(Plugin.prototype);
-BroccoliMergeTrees.prototype.constructor = BroccoliMergeTrees;
-function BroccoliMergeTrees(inputNodes, options) {
-  if (!(this instanceof BroccoliMergeTrees)) return new BroccoliMergeTrees(inputNodes, options);
-  options = options || {};
-  var name = 'broccoli-merge-trees:' + (options.annotation || '');
-  if (!Array.isArray(inputNodes)) {
-    throw new TypeError(name + ': Expected array, got: [' + inputNodes +']');
-  }
-  Plugin.call(this, inputNodes, {
-    persistentOutput: true,
-    needsCache: false,
-    annotation: options.annotation
-  });
-  this.inputNodes = inputNodes;
-  this.options = options;
-}
+const Plugin = require('broccoli-plugin');
+const MergeTrees = require('merge-trees');
 
-BroccoliMergeTrees.prototype.build = function() {
-  if (this.mergeTrees == null) {
-    // Defer instantiation until the first build because we only
-    // have this.inputPaths and this.outputPath once we build.
-    this.mergeTrees = new MergeTrees(this.inputPaths, this.outputPath, {
-      overwrite: this.options.overwrite,
-      annotation: this.options.annotation
+class BroccoliMergeTrees extends Plugin{
+  constructor (inputNodes, options = {}) {
+    if (!Array.isArray(inputNodes)) {
+      throw new TypeError(name + ': Expected array, got: [' + inputNodes +']');
+    }
+    super(inputNodes, {
+      persistentOutput: true,
+      needsCache: false,
+      annotation: options.annotation
     });
+
+    let name = 'broccoli-merge-trees:' + (options.annotation || '');
```

Comment:
Hmm, this seems like it should be above the `if (!Array.isArray` check (right at the beginning of the constructor) since `name` is used there for the error message.